src/main/java/uk/ac/ebi/ampt2d/services/FileTrackingService.java

```diff
@@ -44,29 +44,42 @@
 
     @Autowired
     public FileTrackingService(FileMetadataRepository fileMetadataRepository, FileSystemStorageService
-            fileSystemStorageService){
+            fileSystemStorageService) {
         this.fileMetadataRepository = fileMetadataRepository;
         this.fileSystemStorageService = fileSystemStorageService;
     }
 
     public FileMetadata addFile(InputStream inputStream) throws StorageException, IOException {
         SourceFilePath sourceFilePath = fileSystemStorageService.store(inputStream);
         String fileHash = null;
+        FileMetadata storedFileMetadata = null;
         try {
+            storedFileMetadata = storeTemporalFileMetadata(sourceFilePath);
             fileHash = fileSystemStorageService.getFileHash(sourceFilePath);
-            long fileSize = fileSystemStorageService.getFileSize(sourceFilePath);
-
-            FileMetadata fileMetadata = new FileMetadata(fileHash, FileType.BINARY, fileSize);
-            fileMetadata.addSourceFilePaths(sourceFilePath);
-            return fileMetadataRepository.save(fileMetadata);
-        } catch (IOException e){
-            fileSystemStorageService.delete(sourceFilePath);
+            storedFileMetadata.setHash(fileHash);
+            return fileMetadataRepository.save(storedFileMetadata);
```

Comment:
Maybe these 2 lines could be extracted to a single method like `updateFileMetadataHash`? That would make even more straightforward to follow the 3-step process:

1. Store temporary file metadata
2. Calculate file hash
3. Update file metadata hash